apps/server/src/modules/provisioning/strategy/schulconnex/schulconnex.strategy.ts

```diff
@@ -17,19 +20,25 @@ import {
 } from './service';
 
 @Injectable()
-export abstract class SchulconnexProvisioningStrategy extends ProvisioningStrategy {
+export class SchulconnexProvisioningStrategy extends SchulconnexFetchStrategy {
```

Comment:
This naming convention feels a little misleading. 
- We now have an **SchulconnexAsyncProvisioningStrategy**, so **SchulconnexProvisioningStrategy** sounds like it's the Baseclass
- Additionally **SchulconnexFetchStrategy** also sounds like a subclass of **SchulconnexProvisioningStrategy**

I propose that we rename the _old_ Strategy to **SchulconnexSyncProvisioningStrategy** and the abstract class to **SchulconnexBaseProvisioningStrategy** to make the fork clear.